### ISSUE TITLE
Unable to save changes to my main save file

### ISSUE THREAD

**Issue by Darkminer86** [goatfungus/NMSSaveEditor#394]
whenever I make changes to any of my save files, it works fine. HOWEVER, changing things on my main save and clicking save causes nothing to happen. The last saved date doesn't change, and when I try to close the editor or swap files, I get the warning saying i have unsaved changes and it asks me if I want to save them. Once again, clicking yes/save on that pop up does nothing. How do I fix this issue? I've attached a video which shows the issue. (Yes I know the game is running, and I do not have any save files loaded. The issue still persists when the game is not running)

https://user-images.githubusercontent.com/74679381/120386047-c0a4de00-c2ed-11eb-8d09-9be4408f551e.mp4

**Comment by goatfungus** [comment]
Can you attach a copy of your log file?

**Comment by Darkminer86** [comment]
[NMSSaveEditor.log](https://github.com/goatfungus/NMSSaveEditor/files/6589040/NMSSaveEditor.log)
I don't know if this is the right file but I think it is.

**Comment by Darkminer86** [comment]
@goatfungus Hi! It's been a while since I sent the log file and I still have heard no response... Is there anything wrong with the log file?

**Comment by Biohazard1247** [comment]
I have the same issue and the same message appears in my log as well. Is there still no fix?


Exception in thread "AWT-EventQueue-0" java.lang.ArrayIndexOutOfBoundsException: -1

**Comment by mdzierzkowski** [comment]
Same for me:
```
[INFO] Formatting JSON...
[STDERR] Exception in thread "AWT-EventQueue-1" java.lang.ArrayIndexOutOfBoundsException: -1
[STDERR] 	at java.util.ArrayList.elementData(ArrayList.java:424)
[STDERR] 	at java.util.ArrayList.get(ArrayList.java:437)
```

**Comment by Finnatix21** [comment]
I am having this issue too please help

**Comment by goatfungus** [comment]
This should be fixed in version 1.9.2.
